flocker/control/test/test_httpapi.py

```diff
@@ -440,6 +442,289 @@ def created(ignored):
+            hostname=self.NODE_A,
+            applications=frozenset(),
+            other_manifestations=frozenset([expected_manifestation])
+        )
+        node_b = Node(hostname=self.NODE_B)
+        deployment = Deployment(nodes=frozenset([node_a, node_b]))
+        saving = self.persistence_service.save(deployment)
+
+        def saved(ignored):
+            creating = self.assertResult(
+                b"POST",
+                b"/configuration/datasets/%s" % (
+                    expected_manifestation.dataset.dataset_id.encode('ascii')
+                ),
+                {u"primary": self.NODE_B},
+                INTERNAL_SERVER_ERROR,
+                u'ELIOT LOG REFERENCE'
+            )
+            return creating
+        saving.addCallback(saved)
+        return saving
+    test_only_replicas.todo = (
+        "XXX: Perhaps this test isn't necessary. "
+        "There should always be a primary."
+        "But perhaps there should be a test that demonstrates the general 500 "
+        "response message format."
+        "See https://clusterhq.atlassian.net/browse/FLOC-1393"
+    )
+
+    def test_primary_invalid(self):
+        """
+        A request with an invalid (non-IPv4) primary IP address is rejected
+        with ``BAD_REQUEST``.
+        """
+        expected_manifestation = _manifestation()
+        node_a = Node(
+            hostname=self.NODE_A,
+            applications=frozenset(),
+            other_manifestations=frozenset([expected_manifestation])
+        )
+        deployment = Deployment(nodes=frozenset([node_a]))
+        saving = self.persistence_service.save(deployment)
+
+        def saved(ignored):
+            creating = self.assertResponseCode(
+                b"POST",
+                b"/configuration/datasets/%s" % (
+                    expected_manifestation.dataset.dataset_id.encode('ascii')
+                ),
+                {u"primary": u'192.0.2.257'},
+                BAD_REQUEST,
+            )
+            return creating
+        saving.addCallback(saved)
+        return saving
+
+    def test_no_primary(self):
+        """
+        An update request without a primary address is allowed.
+        """
+        expected_manifestation = _manifestation()
+        node_a = Node(
+            hostname=self.NODE_A,
+            applications=frozenset(),
+            other_manifestations=frozenset([expected_manifestation])
+        )
+        deployment = Deployment(nodes=frozenset([node_a]))
+        saving = self.persistence_service.save(deployment)
+
+        def saved(ignored):
+            creating = self.assertResponseCode(
+                b"POST",
+                b"/configuration/datasets/%s" % (
+                    expected_manifestation.dataset.dataset_id.encode('ascii')
+                ),
+                {},
+                OK,
+            )
+            return creating
+        saving.addCallback(saved)
+        return saving
+    test_no_primary.todo = (
```

Comment:
File a follow up issue? Perhaps the "and you can update metadata and size" issue.